### PR TITLE
Avnet mt3620 sk rev2 hw files

### PR DESCRIPTION
**Pull request by bawilless**
Updates for the Avnet Hardware . . . 

Avnet is releasing new revisions of the MT3620 Module and the Avnet Starter Kit.  Both new boards will be revision 2.  I've generated the required files to support these boards in the SDK *.json and *.h files, and created new directories with the required sample_appliance files under the recently renamed HardwareDefinitions directories.  These boards are available now (8/31/20) so the sooner these can get into the SDK and samples repo, the better.  

We'll be publishing a blog soon to introduce the new Starter Kit and will include details on how to update the SDK hardware directories for the new boards.  Hopefully, this will get us over the gap until a new SDK can be released with minimal issues from the field.

Please let me know if there are any questions.

New files for the SDK installer

New Revision 2 Avnet Azure Sphere COTS Module: avnet_aesms_mt3620_rev2
- Added avnet_aesms_mt3620_rev2.json and \inc\hw\avnet_aesms_mt3620_rev2.h files

New Revision 2 Avnet Azure Sphere Starter Kit: avnet_mt3620_sk_rev2
- Added avnet_mt3620_sk_rev2.json and \inc\hw\avnet_mt3620_sk_rev2.h files

New hardware directory for the azure-sphere-samples\HardwareDefinitions directories
- New avnet_mt3620_sk_rev2 directory with avnet_mt3620_sk_rev2\sample_appliance.json and \inc\hw\sample_appliance.h files

Updated avnet_mt3620_sk files to define the SAMPLE_NRF_* signals on the PMOD connector since that's the only place where CTS and RTS are available on the kit

Updated all Avnet files that reference the Click Board Sockets, this was "SOCKET" changed to "CLICK"

### CODE REVIEW

File: HardwareDefinitions/avnet_mt3620_sk/sample_appliance.json [hunk 1]
```diff
@@ -25,19 +25,19 @@
...
-        {"Name": "SAMPLE_NRF52_UART", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU1_UART", "Comment": "MT3620 SK: Connect external NRF52 UART using SOCKET1: \"MISO\" (RX), \"SCK\" (TX), \"CS\" (CTS), and \"MOSI\" (RTS)."},
-        {"Name": "SAMPLE_DEVICE_STATUS_LED", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_GPIO0", "Comment": "MT3620 SK: Connect external red LED using SOCKET1, pin PWM."},
-        {"Name": "SAMPLE_PENDING_UPDATE_LED", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_GPIO2", "Comment": "MT3620 SK: Connect external blue LED using SOCKET1, pin PWM2."},
+        {"Name": "SAMPLE_NRF52_RESET", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_GPIO17", "Comment": "MT3620 SK: Connect external NRF52 RESET signal using the PMOD connector: Pin-8."},
+        {"Name": "SAMPLE_NRF52_DFU", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_GPIO2", "Comment": "MT3620 SK: Connect external NRF52 DFU signal using the PMOD connector: Pin-7"},
+        {"Name": "SAMPLE_NRF52_UART", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU0_UART", "Comment": "MT3620 SK: Connect external NRF52 UART using the PMOD connector): (RX Pin-3), (TX Pin-2), (CTS Pin-1), and (RTS Pin-4)."},
```
Comment (anpintil) [inline HardwareDefinitions/avnet_mt3620_sk/sample_appliance.json:38]: Why were these changes necessary? We should talk about testing the changes. We will be in contact on a private thread.

File: HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {"Name": "SAMPLE_RGBLED_RED", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_USER_LED_RED", "Comment": "MT3620 SK: User LED RED Channel."},
+        {"Name": "SAMPLE_RGBLED_GREEN", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_USER_LED_GREEN", "Comment": "MT3620 SK: User LED GREEN Channel."},
+        {"Name": "SAMPLE_RGBLED_BLUE", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_USER_LED_BLUE", "Comment": "MT3620 SK: User LED BLUE Channel."},
+        {"Name": "SAMPLE_UART_LOOPBACK", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU0_UART", "Comment": "MT3620 SK: Connect CLICK1 RX (RX) to CLICK1 TX (TX)."},
+        {"Name": "SAMPLE_LSM6DS3_I2C", "Type": "I2cMaster", "Mapping": "AVNET_MT3620_SK_ISU2_I2C", "Comment": "MT3620 SK: Connect external LSM6DS3 to I2C using CLICK1, pin MISO (SDA) and pin MOSI (SCL)."},
+        {"Name": "SAMPLE_LSM6DS3_SPI", "Type": "SpiMaster", "Mapping": "AVNET_MT3620_SK_ISU1_SPI", "Comment": "MT3620 SK: Connect external LSM6DS3 to SPI using CLICK1, pin MISO (MISO), pin SCK (SCLK), pin MOSI (MOSI) and CLICK2 pin CS (CSB)."},
```
Comment (anpintil) [inline HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json:35]: The AVNET_MT3620_SK_ISU1_SPI isn't defined in the the avnet_mt3620_sk_rev2.json file. Should the SAMPLE_LSM6DS3_SPI be mapped to AVNET_MT3620_SK_ISU1_SPI_CONFIGURABLE_CLICK2 instead and the comment changed accordingly?

File: HardwareDefinitions/avnet_mt3620_sk/sample_appliance.json (hunk 1 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_mt3620_sk/sample_appliance.json:38]: HI Andrea,

Thanks for digging in to these changes and asking questions.  

There are two logical changes here for the original Starter Kit files, first SAMPLE_LSM6DS3_I2C was changed from ISU1 I2C to ISU2 I2C because ISU I2C signals are not present on either click sockets; so that was never correct.  ISU1 I2C is present on the click sockets so we define that interface for correctness.

The second set of changes for the NRF external update example originally defined the IUS1 UART on the click socket.  While the ISU1 UART TX and RX signals are available on the click sockets, the ISU UART RTS and CTS signals are not.  The only place UART RTS and CTS signals are available on the Starter Kit are on the PMOD connector site which uses ISU0.

I did test the NRF external update changes on my starter kit using the proposed changes and the PMOD signals.  I don't have the LSM6DS3 hardware to test the ISU change, so I did not test that change with the example software.  

Note that the NRF external update instructions include a wiring diagram for the Seeed Development board, there are not any diagrams for wiring the Avnet Starter Kit, so this change does not drive any documentation changes.  Once these changes are accepted and we release the Starter Kit Revision 2, I'll update and submit a pull request to update the documentation for the NRF external update example and add wiring diagrams for both kits.

You can review these interfaces in the Starter Kit Rev1 User Guide here:  https://www.avnet.com/opasdata/d120001/medias/docus/197/Azure%20Sphere%20Starter%20Kit%20User%20Guide%20(v1.5).pdf

Brian

File: HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json (hunk 2 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json:35]: Good catch.  SAMPLE_LSM6DS3_SPI should be mapped to AVNET_MT3620_SK_ISU0_SPI, not ISU1.  I'll update the sample_appliance.* files and update the pull request.

Comment (bawilless) [reply HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json]: My branch has been updated with this change.  Please let me know if there are any other questions.

File: HardwareDefinitions/avnet_mt3620_sk/sample_appliance.json (hunk 1 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_mt3620_sk/sample_appliance.json:38]: A colleague just corrected me on the SAMPLE_LSM6DS3_I2C change I made for the Starter Kit Rev1.  I was incorrect in my statement and should have never changed that definition.  I've made the correction in the branch.

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    "MainCoreHeaderFileTopContent": [
+      "/* Copyright (c) Microsoft Corporation. All rights reserved.",
+      "   Licensed under the MIT License. */",
+      "",
+      "// This JSON hardware definition file specifies peripheral pinouts",
+      "// of the Rev.2 Avnet MT3620 Starter Kit (SK2). The Rev.2 board can",
+      "// be fitted with an Ethernet Click, it also allows reconfiguration",
+      "// of Click2, Pmod and Grove connectors, by moving config resistors.",
+      "// See Hardware User Guide (pages 22-26) for more details.",
+      "// http://avnet.me/mt3620-kit-UG-V2 ",
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json:17]: Is the link correct?

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+// ISU1 maps also to the PMOD connector (unpopulated). It's default
+// configuration supports type 4A UART Pmod peripheral boards for
+// BLE or wired UART expansion. Alternatively it can be configured
+// to support type 2A SPI Pmod peripheral boards.
+// UART configuration (default) has R39 and R50 populated with
+// zero ohm resistors (and locations R41 and R52 unpopulated).
+// SPI configuration requires de-populating R39 and R50, then
+// moving these config resistors to locations R41 and R52.
+//
+// ISU2 maps to the Grove connector. It�s default configuration
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h:27]: It�s - typo?

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json (hunk 3 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json:17]: Yes, this is a short URL that we can change on the backend anytime.  We will update the final page when we release the new hardware and corresponding documentation.

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h (hunk 4 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h]: I've replaced this character with a valid " ' " character, regenerated the *.h file and pushed the changes to my branch.

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json [hunk 5]
```diff
@@ -0,0 +1,90 @@
+{
+  "Metadata": {
+    "Type": "Azure Sphere Hardware Definition",
+    "Version": 1
+  },
+  "Description": {
+    "Name": "Avnet MT3620 Starter Kit (SK) Rev2",
+    "MainCoreHeaderFileTopContent": [
+      "/* Copyright (c) Microsoft Corporation. All rights reserved.",
+      "   Licensed under the MIT License. */",
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json:10]: We should also update the license, now that AVNET owns the hardware definitions.

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_aesms_mt3620_rev2.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
+{
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_aesms_mt3620_rev2.json:1]: Can you please move this to avnet_mt3620_sk_rev2?

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json [hunk 7]
```diff
@@ -0,0 +1,90 @@
+{
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json:1]: Can you please move this to avnet_mt3620_sk_rev2?

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_aesms_mt3620_rev2.h [hunk 8]
```diff
@@ -0,0 +1,131 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_aesms_mt3620_rev2.h:1]: Can you please move this to avnet_mt3620_sk_rev2/inc/hw?

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h [hunk 9]
```diff
@@ -0,0 +1,155 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (anpintil) [inline HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h:1]: Can you please move this to avnet_mt3620_sk_rev2/inc/hw?

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,5 @@
+﻿################################################################################
```
Comment (anpintil) [inline .gitignore:1]: Can you please remove this file?

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json (hunk 5 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json:10]: I'll make this change for ALL the avnet specific hardware files.  Is that what you're asking?

Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json]: Done

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_aesms_mt3620_rev2.json (hunk 6 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_aesms_mt3620_rev2.json]: Done

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json (hunk 7 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/avnet_mt3620_sk_rev2.json]: Done

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_aesms_mt3620_rev2.h (hunk 8 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_aesms_mt3620_rev2.h]: Done

File: HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h (hunk 9 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_sk_rev2_files_for_SDK_installer/inc/hw/avnet_mt3620_sk_rev2.h]: Done

File: .gitignore (hunk 10 above)
Comment (bawilless) [reply .gitignore:1]: Done

File: HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+        {"Name": "SAMPLE_NRF52_UART", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU0_UART", "Comment": "MT3620 SK: Connect external NRF52 UART using CLICK1: RX (RX), TX (TX), CTS (CS), and RTS (SDI)."},
+        {"Name": "SAMPLE_STM32_UART", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU0_UART", "Comment": "MT3620 SK: Connect external STM32 UART using CLICK1: RX (RX), TX (TX), CTS (CS), and RTS (SDI)."},
+        {"Name": "SAMPLE_DEBUG_UART", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU1_UART", "Comment": "MT3620 SK: Connect external STM32 UART using CLICK2: RX (RX), TX (TX), CTS (CS), and RTS (SDI)."},
+        {"Name": "SAMPLE_DEVICE_STATUS_LED", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_GPIO0", "Comment": "MT3620 SK: Connect external red LED using CLICK1, pin PWM."},
+        {"Name": "SAMPLE_PENDING_UPDATE_LED", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_APP_STATUS_LED_YELLOW", "Comment": "MT3620 SK: User APP LED (Yellow)."},
+        {"Name": "SAMPLE_LSM6DSO_I2C", "Type": "I2cMaster", "Mapping": "AVNET_AESMS_ISU2_I2C", "Comment": "MT3620 SK: LSM6DSO accelerometer."}
```
Comment (anpintil) [inline HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json:44]: Shouldn't this be AVNET_MT3620_SK_ISU2_I2C?

File: .gitignore [hunk 12]
```diff
@@ -2,4 +2,4 @@
 # This .gitignore file was automatically created by Microsoft(R) Visual Studio.
 ################################################################################
```
[the remark this answers is not in the document]
Comment (bawilless) [reply .gitignore:3]: Done

File: HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+        {"Name": "SAMPLE_POTENTIOMETER_ADC_CHANNEL", "Type": "int", "Mapping": "MT3620_ADC_CHANNEL1", "Comment": "MT3620 SK: Connect external potentiometer to ADC controller 0, channel 1 using CLICK1 AN. In the app manifest, it is only necessary to request the capability for the ADC Group Controller, SAMPLE_POTENTIOMETER_ADC_CONTROLLER."},
+        {"Name": "SAMPLE_RGBLED_RED", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_USER_LED_RED", "Comment": "MT3620 SK: User LED RED Channel."},
+        {"Name": "SAMPLE_RGBLED_GREEN", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_USER_LED_GREEN", "Comment": "MT3620 SK: User LED GREEN Channel."},
+        {"Name": "SAMPLE_RGBLED_BLUE", "Type": "Gpio", "Mapping": "AVNET_MT3620_SK_USER_LED_BLUE", "Comment": "MT3620 SK: User LED BLUE Channel."},
+        {"Name": "SAMPLE_UART_LOOPBACK", "Type": "Uart", "Mapping": "AVNET_MT3620_SK_ISU0_UART", "Comment": "MT3620 SK: Connect CLICK1 RX (RX) to CLICK1 TX (TX)."},
+        {"Name": "SAMPLE_LSM6DS3_I2C", "Type": "I2cMaster", "Mapping": "AVNET_MT3620_SK_ISU2_I2C", "Comment": "MT3620 SK: Connect external LSM6DS3 to I2C using CLICK1, pin MISO (SDA) and pin MOSI (SCL)."},
```
[the remark this answers is not in the document]
Comment (bawilless) [reply HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json]: You are looking at an outdated file.  Please update and you'll see that SAMPLE-SLM6DS3_I2C is mapped to ISU1

{"Name": "SAMPLE_LSM6DS3_I2C", "Type": "I2cMaster", "Mapping": "AVNET_MT3620_SK_ISU1_I2C", "Comment": "MT3620 SK: Connect external LSM6DS3 to I2C using CLICK1, pin MISO (SDA) and pin MOSI (SCL)."},

File: HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json (hunk 11 above)
Comment (bawilless) [reply HardwareDefinitions/avnet_mt3620_sk_rev2/sample_appliance.json:44]: You are correct, I skipped a level of indirection.  Ha!  This change has been checked-in.

**Comment by anpintil** [comment]
We will add the new board/chip files in the next SDK release. I think the best way to move forward is to remove those files from this PR and complete the PR only with the sample files. Does this approach work for you?

**Comment by bawilless** [comment]
Just to be clear . . .

You want to remove the files that will be delivered in the SDK from the PR . . .

../HardwareDefinitions/avnet_mt3620_sk_rev2/avnet*.* // Keep only sample_appliance.json
../HardwareDefinitions/avnet_mt3620_sk_rev2/inc/hw/avnet*.* // keep only sample_appliance.h

You will extract these files and include them in the next SDK installer.
Brian

**Comment by anpintil** [comment]
Yes, that is correct. I pulled the board/chip specific files and we will publish them with the next version of the SDK. The PR should contain only the changes made to sample_appliance.json and  sample_appliance.h (rev1 and rev2).

**Comment by bawilless** [comment]
done

**Comment by bawilless** [comment]
I don't see this PR in the main branch after the 20.10 release.  I did note that the new *_rev2.* files were included in the 20.10 SDK installer.  Do you need anything from me to get these changes into the main branch?

Thanks

**Review by richardtaylorrt — APPROVED** [review]
